sonic_installer/bootloader/aboot.py

```diff
@@ -205,6 +210,22 @@ def verify_next_image(self):
         image_path = os.path.join(self.get_image_path(image), DEFAULT_SWI_IMAGE)
         return self._verify_secureboot_image(image_path)
 
+    def set_fips(self, image, enable):
+        if enable:
+            click.echo('Enabling FIPS...')
+            fips = "1"
+        else:
+            click.echo('Disabling FIPS...')
+            fips = "0"
+        cmdline = self._get_image_cmdline(image)
+        cmdline = re.sub(' sonic_fips=[^\s]', '', cmdline) + " sonic_fips=" + fips
```

Comment:
>re.sub [](http://example.com/codeflow?start=18&length=6)

Suggest parsing it instead of manipulate single command string. You may use `shlex.split`.

Reply:
It is to remove the old sonic-fips option, and then add the new setting, and we do not want to have impact on the other settings.
For example, the options as below:
```
rw console=tty0     console=ttyS0,  9600n8    quiet intel_idle.max_cstate=0 sonic_fips=0 og_size=4096
```
After changed, the options with multiple space characters will not be changed
```
rw console=tty0     console=ttyS0,  9600n8    quiet intel_idle.max_cstate=0 og_size=4096 sonic_fips=1
```